core/modules/filters/is.js

```diff
@@ -27,40 +27,44 @@ Export our filter function
 */
 exports.is = function(source,operator,options) {
 
+	var isFilterOperators = getIsFilterOperators(),
+		subops = operator.operand.split("+"),
+		num_of_subops = subops.length;
 
-	if( !operator.operand) {
-		// Return all tiddlers if the operand is missing
+	//Make sure all the operands are defined.
+	for (var t = 0; t < num_of_subops; t++){
+		if( !isFilterOperators[subops[t]] ) {
+			return [$tw.language.getString("Error/IsFilterOperator")];
+		}
+	}
+
+	if(num_of_subops === 0) { 	// Return all tiddlers if the operand is missing
 		var results = [];
 		source(function(tiddler,title) {
 			results.push(title);
 		});
 		return results;
-	}
+	} else if(num_of_subops === 1) {	// Shortcut the Single Operator
+		var operator = isFilterOperators[subops[0]];
+		return operator(source,operator.prefix,options);
 
-	// Get our isfilteroperators
-	var isFilterOperators = getIsFilterOperators(),
-	    subops = operator.operand.split("+"),
-		filteredResults = {},
-		results = [];
-	for (var t=0; t<subops.length; t++) {
-		var subop = isFilterOperators[subops[t]];
-		if(subop) {
-			filteredResults[subops[t]] = subop(source,operator.prefix,options);
-		} else {
-			return [$tw.language.getString("Error/IsFilterOperator")];
+	} else {	// Handle multiple operators
+		var filtered_results = {},
+			results = [];
+		for(var t=0; t < num_of_subops; t++){
+			var operator = isFilterOperators[subops[t]];
+			operator(source,operator.prefix,options).forEach(function(element) { filtered_results[element] = True});
 		}
-		
+
+		// Sort the output by the input (There may be a better way to do this)
+		source(function(tiddler,title) {
+			if(filtered_results[title] === True) {
```

Comment:
Is "True" correct?

Reply:
It appears to work probably because I'm using the smae value when  I'm setting it in the for loop above, but for clarity's sake, I'll make it explicitly a string.